html/pfappserver/lib/pfappserver/Form/Config/Provisioning/mobileconfig.pm

```diff
@@ -8,27 +8,107 @@ pfappserver::Form::Config::Provisioning - Web form for a switch
 
 =cut
 
+use pf::config;
 use HTML::FormHandler::Moose;
 extends 'pfappserver::Form::Config::Provisioning';
 with 'pfappserver::Base::Form::Role::Help';
 
+has_field 'company' =>
+  (
+   type => 'Text',
+  );
+
+#has_field 'fingerprint' =>
+#  (
+#   type => 'hidden',
+#   value => $data,
+#  )
+
 has_field 'ssid' =>
   (
    type => 'Text',
    label => 'SSID',
   );
 
-has_field 'ca_cert_path' =>
+has_field 'broadcast' =>
+  (
+   type => 'Checkbox',
+   label => 'Broadcast network',
+   value => 'true',
+   checkbox_value => 'false',
+   tags => { after_element => \&help,
+             help => 'Check this box if your network is using a hidden SSID' },
+  );
+
+has_field 'security_type' =>
+  (
+   type => 'Select',
+   multiple => 0,
+   label => 'Security type',
+   options_method => \&option_security,
+   element_class => ['chzn-deselect'],
+   tags => { after_element => \&help,
+             help => 'Select the type of security applied for your SSID' },
+  );
+
+has_field 'eap_type' =>
   (
-   type  => 'Text',
-   label => 'Certificate',
+   type => 'Select',
+   multiple => 0,
+   label => 'EAP type',
+   options_method => \&options_eap_type,
+   element_class => ['chzn-deselect'],
+   tags => { after_element => \&help,
+             help => 'Select the EAP type of your SSID' },
+  );
+
+has_field 'passcode' =>
+  (
+   type => 'Text',
+   label => 'Wifi Key',
+  );
+
+has_field 'reversedns' =>
+  (
+   type => 'Text',
+   label => 'ReverseDNS identifier',
+   tags => { after_element => \&help,
+             help => 'Example : if your dns name is www.packetfence.org it become org.packetfence.www'},
+  );
+
+has_field 'ca_cert' =>
+  (
+   type => 'TextArea',
+   label => 'The base64 Certificate of Authority',
```

Comment:
Wording could be different but people need to copy the content of the certificate in the configuration.
Maybe the label should read "The Certificate of Authority in pem format"
I am opened to suggestions.